sdcm/cluster.py

```diff
@@ -4307,6 +4307,16 @@ def _wait_for_nodes_up_and_normal():
 
         _wait_for_nodes_up_and_normal()
 
+    @retrying(n=360, sleep_time=10, allowed_exceptions=ClusterNodesNotReady, message="Waiting for CQLPing to return")
+    def wait_native_transport(self, node):
```

Comment:
is there a difference from this and `wait_db_up` where we wait for cql port used?

Reply:
also what about using the newly introduced `node.wait_node_fully_start()`

we introduced it for restarts, but the case here is similar..

Reply:
AFAIU, `node.wait_node_fully_start()` by itself won't be enough, as the new node while joining the cluster, will have both jmx and Scylla services running, will report in `nodetool status` it is `UN`, but it won't yet be servicing full I/O.

it could be related to lucky (either good or bad) that the code I run, on a test that is consistently failing on master, now passed with the change in this PR.

about the idea to move it elsewhere, like inside `BaseNode`, i will do the change now and push it to re-review this patch

Reply:
@vponomaryov do you think it could help in operator to wait for node being fully up?

Reply:
I have a better explanation what's going on here:

we don't set the `verification_node` on the `self.cluster.wait_for_nodes_up_and_normal(nodes=[new_node])` call
hence it randomly pick the node to do nodetool status on, and it can be new_node, and why you ask it, it would say UN, while other node won't yet see it as UP, this we for sure.

so calling, `node.wait_node_fully_start()` should do the same trick as what you introduced in this PR 

I would also recommend fixing `wait_for_nodes_up_and_normal` to handle this situation better, and not pick randomly the `verification_node`,  if it's single node to be waited, it should use the same one for verification

I'm guessing since you are running this specific nemesis with 3 node growshrink at a time, this situation just happens more.

Reply:
also `_add_new_node_in_new_dc` should be handle the same as we do in PR (how ever we fix it)

Reply:
@soyacz @fruch , made changes, and now i'm using `wait_node_fully_start()`, and the new function `wait_native_transport()` is now part of it

Reply:
> I have a better explanation what's going on here:
> 
> we don't set the `verification_node` on the `self.cluster.wait_for_nodes_up_and_normal(nodes=[new_node])` call hence it randomly pick the node to do nodetool status on, and it can be new_node, and why you ask it, it would say UN, while other node won't yet see it as UP, this we for sure.
> 
> so calling, `node.wait_node_fully_start()` should do the same trick as what you introduced in this PR
> 
> I would also recommend fixing `wait_for_nodes_up_and_normal` to handle this situation better, and not pick randomly the `verification_node`, if it's single node to be waited, it should use the same one for verification
> 
> I'm guessing since you are running this specific nemesis with 3 node growshrink at a time, this situation just happens more.

i think that is not exactly what happens, because as it was said multiple times, a node being `UN` doesn't mean much... there is also a RFE to add a better way to query if the node is fully serving, or not --> https://github.com/scylladb/scylladb/issues/8275

Reply:
also, added `_add_new_node_in_new_dc()` to use the check `new_node.wait_node_fully_start()`

Reply:
> @vponomaryov do you think it could help in operator to wait for node being fully up?

If operator developers do the same there? If it helps in general, it may help.
But I think they would like to see official and recommended way for getting this info.
In current case it looks like workaround for the absent proper way of checking node readiness.

Reply:
correct, @vponomaryov